app/domain/financial_institutions/repo.py

```
@@ -1,17 +1,44 @@
+        try:
+            self.db.add(model)
+            await self.db.commit()
+            await self.db.refresh(model)
+            return model
+
+        except IntegrityError:
+            await self.db.rollback()
+
+            raise FinancialInstitutionsAlreadyExists(
+                institution=model.name, type=model.type.value
+            )
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)